plugin/core/handlers.py

```diff
@@ -17,9 +17,19 @@ def config(self) -> ClientConfig:
 
     @classmethod
     def instantiate_all(cls) -> 'List[LanguageHandler]':
-        return list(
-            instantiate(c) for c in cls.__subclasses__()
-            if issubclass(c, LanguageHandler))
+        subclasses = []  # type: List[Type[LanguageHandler]]
+        cls.get_final_subclasses(cls.__subclasses__(), subclasses)
+        return list(instantiate(c) for c in subclasses)
+
+    @classmethod
+    def get_final_subclasses(cls, superclasses: 'List[Type[LanguageHandler]]',
+                             results: 'List[Type[LanguageHandler]]') -> None:
```

Comment:
You don't have to quote types anymore.

Reply:
<img width="634" alt="Screenshot 2020-04-29 at 20 46 20" src="https://user-images.githubusercontent.com/153197/80634282-828a1a80-8a5a-11ea-82cf-366418e7b234.png">

Reply:
Oh right, I missed that we're in the LanguageHandler class itself. So we have to use a forward-declaration.